### DISCUSSION TITLE
MotoRos2 installation issue

### DISCUSSION THREAD

**Discussion by rsaintobert** [Yaskawa-Global/motoros2#283] [Troubleshooting] [answered]
I’m having an issue while installing MotoRos2 on a YRC1000 controller: the MotoPlus application does not seem to be started or working properly at the end of the procedure.

The micro ros agent and the debug_listener tool never receive any connection.

The controller is reachable (response to ping) so I don’t think it’s a network issue.

For information, I have already installed MotoRos2 on 3 other YRC1000 controllers successfully in the past few months, so I’m quite confident with the execution of the procedure.

I have however noticed 2 things I’m not sure it has a relation with the problem:

- just after having loaded the MotoPlus APL and initializing the SRAM RAM DRIVE, I get an info message indicating to perform a ‘safety board flash reset’. I executed the procedure twice with and without this reset and I does not seem to change anything
- I properly get the alarm `8013 [0] Missing MotoROS2 cfg file` on reboot in normal operation mode, but when I load the config file I never get the alarm `8001[10] Speed FB enabled, reboot now`

The release used is 0.1.2.
The ROS version used is humble.
I have no logs from the debug_listener because nothing happens.

I have attached the PANELBOX.LOG file and the config file loaded.
[files.zip](https://github.com/user-attachments/files/16439072/files.zip)

**Comment by gavanderhoorn** [comment]
This reminds me of #200, which unfortunately didn't get resolved.

(difference however is there *is* a line under `//MOTOPLUS APP` in the `PANELBOX.LOG` you've shared, so MotoROS2 must have at least started, as that wouldn't be present otherwise)

To skip to the end of #200 (https://github.com/Yaskawa-Global/motoros2/discussions/200#discussioncomment-8172431): @rsaintobert: do you perhaps have a VGA connector kit for your YRC1000?

**Reply by ted-miller** [reply]
@rsaintobert: Could you provide a CMOS.BIN backup of your robot controller? I can quickly check that everything is configured properly.

> when I load the config file I never get the alarm 8001[10] Speed FB enabled, reboot now

I've been seeing some odd behavior with this alarm recently that I can't quite explain. So I'm suspicious about the routine that enables this behavior. Please check these parameters and manually set them if they're not set.
https://wiki.ros.org/motoman_driver/Troubleshooting#Not_receiving_axis_speeds_from_robot_controller
Make sure to reboot after setting these.

**Reply by rsaintobert** [reply]
@gavanderhoorn unfortunately we don't have the VGA connector kit for the YRC1000
@ted-miller I have added the CMOS.BIN file
[CMOS.zip](https://github.com/user-attachments/files/16444228/CMOS.zip)

**Reply by ted-miller** [reply]
It doesn't appear that the config file loaded properly. The CMOS does not have the configuration file included.

Once I loaded the file, it started working. Note that the robot is still configured with the default IP address for LAN2, 192.168.255.1.

**Reply by gavanderhoorn** [reply]
> just after having loaded the MotoPlus APL and initializing the SRAM RAM DRIVE, I get an info message indicating to perform a ‘safety board flash reset’. I executed the procedure twice with and without this reset and I does not seem to change anything

could this be the FSU interfering?

I've had this problem on my YRC1, where nothing would work until I disabled it (temporarily).

**Reply by rsaintobert** [reply]
@ted-miller I have executed the procedure again, I was not sure I had done it to the end during my last try (I did it many times).
So I have have attached the last PANELBOX.LOG and CMOS.BIN with the full procedure executed.

[files.zip](https://github.com/user-attachments/files/16457767/files.zip)

I have noticed that no INIT_ROS job is created after the load of the config file and reboot. I tried to put it manually but I does not change anything.

@gavanderhoorn how do you disable the FSU temporarily ?

**Reply by gavanderhoorn** [reply]
> how do you disable the FSU temporarily ?

it's explained in the [readme](https://github.com/Yaskawa-Global/motoros2?tab=readme-ov-file#dx200-yrc1000-and-yrc1000micro):

> If the controller is configured with the Functional Safety Unit (FSU), then `SAVE DATA CRC CHECK FUNC (FSU)` must be temporarily disabled during the installation procedure.

The instructions following that should explain how to temporarily disable it, and how to re-enable it after installation.

**Reply by rsaintobert** [reply]
@gavanderhoorn ok, I unfortunately already tried it...

**Reply by ted-miller** [reply]
> I never get the alarm 8001[10] Speed FB enabled, reboot now

> I have noticed that no INIT_ROS job is created after the load of the config file and reboot. I tried to put it manually but I does not change anything.

Please note that these things will only happen after the robot successfully connects with the agent PC.

I did notice that in your config file, it's trying to connect on port 9999, rather than the default of 8888.
This is fine. But make sure that the same port is specified when launching the Agent on the PC. 

After adjusting my agent settings to use port 9999, the robot connected just fine. I got the speed-feedback alarm, rebooted, and init_ros was created. So it looks like your robot controller is configured properly. Just need to resolve any networking issues.

**Reply by rsaintobert** [reply, marked as answer]
The issue is solved now, the problem was indeed a networking issue on the PC side.
After inspecting the network (I should have done it before), I saw UDP packets sent by the controller with a payload indicating an attempt to connect to the PC. It turns out that the UDP packets were not allowed on the connected interface (it's a fresh install of Ubuntu 22.04, I don't know why the configuration is done like this). 
I updated the firewalling rules to allow UDP on it and everything is ok now.
Maybe something to add in the troubleshooting guide when no job is automatically created (which may indicate a connectivity issue) ?
Many thanks for your help anyway.

**Reply by gavanderhoorn** [reply]
Thanks @rsaintobert for letting us know.

> Maybe something to add in the troubleshooting guide

yes, this would indeed make a good addition.

We should probably just mention checking `ufw` configuration as a step in the installation and how it could be blocking traffic to the Agent.

**Reply by gavanderhoorn** [reply]
> Maybe something to add in the troubleshooting guide

Tracking in #288.
